vowpalwabbit/parse_regressor.cc

```diff
@@ -294,21 +334,27 @@ void save_load_header(vw& all, io_buf& model_file, bool read, bool text)
                         "", read,
                         "\n", 1, text);
 
-    text_len = sprintf_s(buff, buf_size, "options:%s\n", all.file_options->str().c_str());
-    uint32_t len = (uint32_t)all.file_options->str().length() + 1;
-    memcpy(buff2, all.file_options->str().c_str(), min(len, buf_size));
-
     if (read)
-    { len = buf_size;
+    {
+        uint32_t len;
+        size_t ret = bin_read_fixed(model_file, (char*)&len, sizeof(len), "");
+        if (len > 104857600 /*sanity check: 100 Mb*/ || ret < sizeof(uint32_t))
+            THROW("bad model format!");
+        resize_buf_if_needed(buff2, buf2_size, len);
+        bytes_read_write += bin_read_fixed(model_file, buff2, len, "") + ret;
```

Comment:
I'm worried that the output of this call is not checked as in _validated().  The _validated() functions were added to prevent corrupt model failure modes.   Can you add the same checks in?